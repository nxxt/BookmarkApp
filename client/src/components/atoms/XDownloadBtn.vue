<template>
  <v-tooltip bottom>
    <template v-slot:activator="{ on, attrs }">
      <v-btn
        id="download"
        small
        v-bind="attrs"
        icon
        v-on="on"
        @click="$emit('click')"
      >
        <v-icon>mdi-download</v-icon>
      </v-btn>
    </template>
    <span>download md</span>
  </v-tooltip>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'

@Component
export default class XDownloadBtn extends Vue {}
</script>
