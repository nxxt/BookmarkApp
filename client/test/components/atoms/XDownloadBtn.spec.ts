import Vuetify from 'vuetify'
import { mount, createLocalVue } from '@vue/test-utils'
import XDownloadBtn from '@/components/atoms/XDownloadBtn.vue'

const localVue = createLocalVue()

describe('XDownloadBtn.vue', () => {
  let vuetify: typeof Vuetify
  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const factory = (options?: any) => {
    return mount(XDownloadBtn, {
      localVue,
      vuetify,
      ...options
    })
  }

  // TEST
  describe('template parts', () => {
    it('text', () => {
      const wrapper = factory()
      const btn = wrapper.find('.v-btn')
      expect(btn.attributes().id).toBe('download')
    })
  })

  describe('event', () => {
    const wrapper = factory()

    const event = jest.fn()
    wrapper.vm.$on('click', event)
    expect(event).toHaveBeenCalledTimes(0)

    wrapper.find('.v-btn').trigger('click')

    expect(event).toHaveBeenCalledTimes(1)
  })

  describe('the entire template', () => {
    it('snapshot', () => {
      const wrapper = factory({})
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
