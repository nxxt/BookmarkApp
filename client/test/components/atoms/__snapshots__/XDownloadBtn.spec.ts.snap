// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XDownloadBtn.vue the entire template snapshot 1`] = `"<span class=\\"v-tooltip v-tooltip--bottom\\"><!----><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--small\\" id=\\"download\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\"><span class=\\"v-btn__content\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-download theme--light\\"></i></span></button></span>"`;
